ntloss/core.py

```diff
@@ -234,9 +309,12 @@ def forward(
         softmax_probs = F.softmax(nt_logits, dim=-1)
 
         # compute absolute difference between the true numbers and all possible number values
-        abs_diff = torch.abs(
-            y[valid_positions].unsqueeze(-1) - self.number_values_dense
-        )
+        if self.setup_dist_lookup or isinstance(self.squash_factor, Number):
```

Comment:
here we can then always do the dist lookup. We just need to ensure the devices match, so we have to cast both attributes to the device on which labels live